### ISSUE TITLE
Why doesn't the model input include attention_mask?

### ISSUE THREAD

**Issue by li3cmz** [microsoft/DialoGPT#58]
https://github.com/microsoft/DialoGPT/blob/fa0c0c53a0e6d75b6541e50faa2d77ba480b27d9/LSP_train.py#L281

Since it is a LMHeadModel, the `1^th`-`n^th` tokens are used to predict the `(n+1)^th` token during training, so why not introduce attention_mask for masking the `(n+2)^th`-`(n+m)^th` tokens. Without attention_mask, there maybe an inconsistency between the training and the testing scene. Is it possible to add attention_mask during training to make the testing better?

**Comment by chujiezheng** [comment]
Because GPT is a uni-directional language model. It does not need attention mask.

**Comment by lmrojasb** [comment]
[DialoGPT/data_loader.py](https://github.com/microsoft/DialoGPT/blob/457835e7d8acd08acf7f6f0e980f36fd327ea37c/data_loader.py#L193)

Why is **the response** concatenated to the **input_ids** for both the train and validation datasets?  Would not this create over-fitted models? Would it be possible to somehow mask the response ids?
